.circleci/config.yml

```diff
@@ -0,0 +1,170 @@
+                    echo 'export PATH=$HOME/pmd-bin-5.4.1/bin:$PATH' >> $BASH_ENV
+                    echo 'export PATH=$HOME/bakalint-0.4.0:$PATH' >> $BASH_ENV
+                    echo 'export PATH=$HOME/elm-format-0.18:$PATH' >> $BASH_ENV
+                    echo 'export PATH=$HOME/.local/tailor/tailor-latest/bin:$PATH' >> $BASH_ENV
+                    echo 'export PATH=$HOME/phpmd:$PATH' >> $BASH_ENV
+                    echo 'export R_LIB_USER=$HOME/.RLibrary:$PATH' >> $BASH_ENV
+      - run:
+          name: AptDependencies
+          command:  bash .ci/deps.apt.sh
+      - run:
+          name: GeneralDependencies
+          command: |
+                    gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB
+                    curl -sSL https://get.rvm.io | bash -s stable
+                    source /etc/profile.d/rvm.sh
+                    rvm install 2.2.2
+                    rvm use 2.2.2 --default
+                    ruby -v
+                    gem install bundler
+                    bash .ci/deps.sh
+      - run:
+         name: CabalDependencies
+         command:  bash .ci/deps.cabal.sh
+      - run:
+         name: DependenciesPip
+         command:   |
+                    git clone https://github.com/pyenv/pyenv.git /tmp/pyenv
+                    if [ ! -d "$HOME/.pyenv" ]; then mkdir -p ~/.pyenv ; fi
+                    cp -R /tmp/pyenv/* ~/.pyenv
+                    export PYENV_ROOT="$HOME/.pyenv"
+                    export PATH="$PYENV_ROOT/bin:$PATH"
+                    eval "$(pyenv init -)"
+                    bash .ci/deps.pip.sh
+      - run:
+         name: JavaDependencies
+         command: bash .ci/deps.java.sh
+      - run:
+         name: OpamDependencies
+         command:   bash .ci/deps.opam.sh
+      - run:
+         name: RDependencies
+         command:   bash .ci/deps.r.sh
+      - run:
+         name: CoalaDependencies
+         command:   bash .ci/deps.coala-bears.sh
+      - run:
+          name: Codecov
+          when: always
+          command: codecov
+      - run:
+          name: Coalaci
+          when: always
+          command: coala-ci -L DEBUG
+      - run:
+          name: Setup
+          when: always
+          command: python setup.py bdist_wheel
+      - run:
+          name: AllDependencies
+          when: always
+          command: |
+                      export NVM_DIR="$HOME/.nvm"
+                      [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"
+                      pip install $(ls ./dist/*.whl)"[alldeps]"
+      - run:
+          name: Documents
+          when: always
+          command: |
+                      echo $CIRCLE_NODE_INDEX
+                      if [ $CIRCLE_NODE_INDEX == 1 ] ; then
+                      pip install -r docs-requirements.txt
+                      fi
+      - run:
+          name: SetupDocs
+          when: always
```

Comment:
Remove this line ( when: always). This command shouldn't run if the dependency and environment setup fails.